### PR TITLE
Add MLX backend for OmnilingualASR — wav2vec2 in MLX, 300M/1B/3B/7B

### PR DESCRIPTION
**Pull request by ivan-digital**
> Stacked on top of #199 (CoreML backend). Merge after that lands. Once #199 is in, I'll rebase this onto `main`.

## Summary

Native MLX/Metal port of Meta's Omnilingual ASR (CTC variant). Loads any of the 8 published MLX repos (300M / 1B / 3B / 7B in 4-bit and 8-bit) directly from HuggingFace and runs inference on Apple GPUs. Reference: [`facebookresearch/omnilingual-asr`](https://github.com/facebookresearch/omnilingual-asr) (Apache 2.0), [arXiv 2511.09690](https://arxiv.org/abs/2511.09690).

This completes the dual-backend story for issue #195 — every published Omnilingual variant is now wired through this Swift module:

- CoreML 5 s / 10 s INT8 (300M only) — `OmnilingualASRModel` (#199)
- MLX 4-bit / 8-bit (300M / 1B / 3B / 7B) — `OmnilingualASRMLXModel` (this PR)

## Implementation

`Sources/OmnilingualASR/MLX/`:

- **`OmnilingualMLXConfig`** — variant table:

  | Variant | Layers | dim | Heads | FFN |
  |---|---|---|---|---|
  | 300M | 24 | 1024 | 16 | 4096 |
  | 1B | 48 | 1280 | 20 | 5120 |
  | 3B | 60 | 2048 | 32 | 8192 |
  | 7B | 128 | 2048 | 32 | 8192 |

  Group size 64, bits 4 or 8 (auto-detected from model id).

- **`Wav2Vec2Frontend`** — fairseq2-compatible 7-layer feature extractor (kernels `[10,3,3,3,3,2,2]`, strides `[5,2,2,2,2,2,2]`, 320× downsample → 50 Hz frames at 16 kHz), per-channel `LayerNorm + GELU`, `post_extract_layer_norm`, `model_dim_proj`, weight-normed grouped `Conv1d` positional encoder (kernel 128, groups 16) with even-kernel trim and residual.

- **`Wav2Vec2EncoderLayer`** — pre-norm self-attention + 2-linear FFN. All 4 attention projections (`q/k/v/output_proj`) and both FFN linears are `QuantizedLinear` with separate per-group `scales`/`biases` plus a regular linear bias. Tensor names match fairseq2 exactly: `self_attn.{q,k,v,output}_proj`, `ffn.{inner,output}_proj`, `self_attn_layer_norm`, `ffn_layer_norm`.

- **`Wav2Vec2Encoder`** — stack of N pre-norm layers + final `encoder.layer_norm`.

- **`CTCHead`** — `QuantizedLinear final_proj → 10288 logits`.

- **`OmnilingualMLXWeightLoader`** — loads the single `model.safetensors`, fuses the PyTorch `weight_norm(dim=2)` parameters of the position encoder (`W = g · v / ||v||_per_kernel_position`), and applies all quantised + dense + norm tensors via the existing `MLXCommon` helpers. Conv weight transpose `[Cout, Cin, K] → [Cout, K, Cin]` is handled by the shared `applyConv1dWeights(transpose: true)`.

- **`OmnilingualASRMLXModel`** — top-level loader; `fromPretrained(variant:bits:...)` downloads the right HF repo, parses `tokenizer.model`, builds modules, applies weights, and warms up via `eval(...)`. `transcribeAudio` reuses the CoreML path's reference utterance-level `layer_normalize` and 40 s hard cap, matching Meta's `MAX_ALLOWED_AUDIO_SEC`.

## CLI

```bash
audio transcribe recording.wav --engine omnilingual --backend mlx                              # 300M @ 4-bit (default)
audio transcribe recording.wav --engine omnilingual --backend mlx --variant 1B                  # 1B @ 4-bit
audio transcribe recording.wav --engine omnilingual --backend mlx --variant 3B --bits 8         # 3B @ 8-bit
audio transcribe recording.wav --engine omnilingual --backend mlx --variant 7B                  # 7B @ 4-bit
```

## Verified output

End-to-end on the 300M-4bit MLX repo, transcripts essentially identical to the CoreML 300M model:

| Lang | Transcript (MLX 300M-4bit) |
|---|---|
| EN | `can you guarantee that the replacement part will be shiped tomorrow` |
| AR | `كما أثنى الزملاء المصارع على لونا` ✓ |
| HI | `लूना को साथी पहलवानों ने भी सरधांजली दी` ✓ Devanagari |
| FR | `pensez à létineraire deski comme unétineraire de rendonner similaire` |

The 1B-4bit variant also loads end-to-end and produces a recognisable English transcript (`"can you guarantee that the replacement par o be shipe tomorrow"` — slightly more 4-bit quant degradation than 300M, expected).

## Tests

- **9 unit tests** (CI): 6 `OmnilingualMLXConfigTests` (variant table + model id resolution) + 3 `Wav2Vec2FrontendShapesTests` (encoder output length math against published shapes for 5 s / 10 s / under-window inputs)
- **7 E2E tests** on 300M-4bit (`E2EOmnilingualMLXTests`): load, warmup, English transcript, 40 s cap rejection, Arabic / Hindi / French FLEURS clips
- **1 E2E test** on 1B-4bit (`E2EOmnilingualMLX1BTests`): exercises larger encoder code path
- **6 new CLI parser tests** in `TranscribeCommandTests` for `--backend mlx` / `--variant` / `--bits`
- **Full unit regression sweep**: 763 tests pass, 0 failures (+20 vs the CoreML PR baseline)

## Test plan

- [x] `swift build` green
- [x] `swift test --filter OmnilingualMLXConfigTests` — 6/6 pass
- [x] `swift test --filter Wav2Vec2FrontendShapesTests` — 3/3 pass
- [x] `swift test --filter TranscribeCommandTests` — 22/22 pass (6 new)
- [x] `swift test --filter E2EOmnilingualMLXTests` — 7/7 pass (real 300M weights, ~5 s wall time)
- [x] `swift test --filter E2EOmnilingualMLX1BTests` — 1/1 pass (real 1B weights, ~56 s including 549 MB download)
- [x] `swift test --skip E2E` — 763/763 pass, 12 skipped, 0 failures
- [x] Multilingual transcripts verified locally for EN / HI / FR / AR
- [ ] CI green
- [ ] 3B / 7B variants verified manually (deferred — same code path, just larger; will spot-check via CLI before release)

### CODE REVIEW

**Comment by ivan-digital** [comment]
Superseded by #201 — single consolidated PR that bundles the CoreML backend (#199) with this MLX work and the  /  extractions. All content from this PR is preserved in #201 as the second commit.
